libraries/AP_PiccoloCAN/AP_PiccoloCAN.cpp

```diff
@@ -690,22 +691,28 @@ bool AP_PiccoloCAN::handle_esc_message(AP_HAL::CANFrame &frame)
         // There are no common error bits between the Gen-1 and Gen-2 ICD
     } else if (decodeESC_StatusBPacket(&frame, &esc.voltage, &esc.current, &esc.dutyCycle, &esc.escTemperature, &esc.motorTemperature)) {
         
-        AP_ESC_Telem_Backend::TelemetryData t {
-            .temperature_cdeg = int16_t(esc.escTemperature * 100),
-            .voltage = float(esc.voltage) * 0.01f,
-            .current = float(esc.current) * 0.01f,
-        };
+        AP_ESC_Telem_Backend::TelemetryData telem = {};
 
-        update_telem_data(addr, t,
+        telem.voltage = float(esc.voltage) * 0.01f;
+        telem.current = float(esc.current) * 0.01f;
+        telem.motor_temp_cdeg = int16_t(esc.motorTemperature * 100);
```

Comment:
You're updating three fields - voltage, current and motor_temp_cdeg, but in the update call you're saying that CURRENT, VOLTAGE and TEMPERATURE are updated.  That means the `update_telem_data` code will be looking in the `temperature_cdeg` field for information *not* in `motor_temp_cdeg`.

If I'm right then the motor temperature reported here will always be zero.  We really should have a simulator to make sure this sort of thing continues to function at a basic level....